### ISSUE TITLE
Don't log user in immediately after password reset?

### ISSUE THREAD

**Issue by barosenfeldSA** [AzureAD/microsoft-authentication-library-for-js#5012]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.22.0

### Wrapper Library

MSAL React (@azure/msal-react)

### Wrapper Library Version

1.4.3

### Description

I have setup B2C authentication in my React app that was created with create-react-app. Everything seems to be working. However, after a user successfully resets their password the _LOGIN_SUCCESS_ event is emitted and the user is automatically logged into the app. Is this the expected behavior or is there something in my code that needs to be changed? I would expect that the user should be navigated back to the login page to login with their new password.

### MSAL Configuration

```javascript
{
        auth: {
            authority: `https://${tenantName}.b2clogin.com/${tenantId}/${signInSignUpPolicy}`,
            knownAuthorities: [`${tenantName}.b2clogin.com`],
            clientId: authConfig.clientId,
            postLogoutRedirectUri: window.location.origin,
            redirectUri: window.location.origin,
        },
        cache: {
            cacheLocation: 'localStorage',
            storeAuthStateInCookie: false,
        },
        forgotPasswordRequest: { authority: `https://${tenantName}.b2clogin.com/${tenantId}/${pwdResetPolicy}` },
}
```

### Relevant Code Snippets

```javascript
export function App() {
    /* other logic */

    return (
       <...other wrapper components>
          <MsalWrapper>
              <Authenticate>
                 <Banner />
                 <AppContent />
              </Authenticate>
          </MsalWrapper>
       </...other wrapper components>
    );
}

function MsalWrapper({ children }) {
    const [msalInstance, setMsalInstance] = useState();
    const authentication = useSelector((state) => state.authentication);

    useEffect(() => {
        if (authentication) {
            /* Need to wait until authentication has been fetched for the current tenant before 
             * initializing PCA.
             */
            const msalConfig = getMsalConfig(authentication);
            const msalInstance = new PublicClientApplication(msalConfig);

            setMsalInstance(msalInstance);
        } else {
            setMsalInstance(undefined);
        }
    }, [authentication]);

    if (!msalInstance) {
        return null;
    }

    return <MsalProvider instance={msalInstance}>{children}</MsalProvider>;
}


function Authenticate (props) {
    const { instance } = useMsal();
    const account = instance.getActiveAccount();
    const tenantAuthConfig = useSelector((state) => state?.authentication)
    const [userProfileMapping, setUserProfileMapping] = useState();

    const logout = () => {
        instance.handleRedirectPromise().then(() => instance.logoutRedirect())
    }

    useEffect(() => {
        const passwordResetURL = getMsalConfig(tenantAuthConfig).forgotPasswordRequest;
        const callbackId = instance.addEventCallback(async(event) => {
            if (event?.eventType === EventType.LOGIN_SUCCESS) {
                instance.setActiveAccount(event.payload.account)
            } else if (event?.eventType === EventType.LOGIN_FAILURE && event?.interactionType === InteractionType.Redirect) {
                if (event?.error?.errorMessage?.includes('AADB2C90118')) {
                    return instance.loginRedirect(passwordResetURL);
                }
            }
        })
        return () => {
            instance.removeEventCallback(callbackId)
        }
    }, [])

    const getAccessToken = useCallback(
        async (scopes) =>
            instance
                ? instance
                      .acquireTokenSilent({ scopes, account })
                      .then((response) => {
                          return response.accessToken;
                      })
                      .catch((error) => {
                          if (error instanceof InteractionRequiredAuthError) {
                              // fallback to interaction when silent call fails
                              return instance.acquireTokenRedirect({ scopes, account });
                          }
                      })
                : null,
        [instance],
    );

    //Custom functionality to map AzureAD account to custom user profile object
    useEffect(() => {
        async function getUserProfile() {
            const userId = account?.localAccountId;
            const authToken = await getAccessToken(scopes);
            const response = /* ...await API call */
            setUserProfileMapping(response.data);
        }

        if (instance) {
            if (account?.localAccountId) {
                getUserProfile();
            } else {
                setUserProfileMapping(undefined);
            }
        }
    }, [account && account.localAccountId]);

    return (
        <MsalAuthenticationTemplate interactionType='redirect'>
            {/* Custom AuthenticationContext */}
            <AuthenticationContext.Provider
                value={{
                    account: { ...account, name: account?.name ?? `${account?.idTokenClaims?.given_name} ${account?.idTokenClaims?.family_name}` },
                    logout,
                    getAccessToken,
                    authorizationScopes: scopes,
                    userProfile: userProfileMapping,
                    updateUserProfile: (updatedProfile) => {
                        setUserProfileMapping(updatedProfile);
                    },
                }}
            >
                {props.children}
            </AuthenticationContext.Provider>

        </MsalAuthenticationTemplate>
    )
}
```

### Identity Provider

Azure B2C Basic Policy

### Source

External (Customer)

**Comment by sameerag** [comment]
@derisen Can you please confirm this behavior once with our samples? I assume the general behavior has to be re-enter the password.

**Comment by derisen** [comment]
@BeeR-II this is expected, with both legacy and self-service password reset user flows. If you want the user to sign-in again with their new password, you can listen for the `LOGIN_SUCCESS` event, and then check the `idTokenClaims `of the user. If the `isForgotPassword ` claim appears and is set to true, you can log them out and ask them to sign-in again -you might need to migrate to [self-service password reset flow](https://docs.microsoft.com/en-us/azure/active-directory-b2c/add-password-reset-policy?pivots=b2c-user-flow) for that, which is also easier to handle. If you choose not to migrate, you can check for the `acr`/`tfp` claim and see if it's set to your password reset policy name (e.g. `b2c_1_reset`). Would that help?

**Comment by barosenfeldSA** [comment]
@derisen Thank you for the quick response and confirmation. That was what I had implemented as a workaround in the meantime, but I'm happy to know that this is the expected behavior and that is the recommended approach. 

For the benefit of anyone else reading this thread, after listening for the `LOGIN_SUCCESS` event and triggering log out if the the user was resetting the password, I was still seeing the app contents flash on the screen for a second between password reset and logout. This was solved by conditionally rendering the children of the `MsalAuthenticationTemplate`, only if account is defined.
